sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/randomExpressions.scala

```diff
@@ -47,10 +46,8 @@ abstract class RDG extends UnaryExpression with ExpectsInputTypes with Stateful
   override def seedExpression: Expression = child
 
   @transient protected lazy val seed: Long = seedExpression match {
-    case Literal(s, IntegerType) => s.asInstanceOf[Int]
-    case Literal(s, LongType) => s.asInstanceOf[Long]
-    case _ => throw new AnalysisException(
-      s"Input argument to $prettyName must be an integer, long or null literal.")
+    case e if e.dataType == IntegerType => e.eval().asInstanceOf[Int]
+    case e if e.dataType == LongType => e.eval().asInstanceOf[Long]
```

Comment:
Hm, I remember Scala 2.13 issues a warning about it when there's no default case from case match. Can we still add `case _ =>`?

Reply:
Oh, it's merged already. NVM, it's a nit and will be fixed later in batch together.

Reply:
Oh, the scala 2.13 build is not checked for a while because GA is broken. Let's fix them in a batch.

Reply:
ah, get it. Will correct this next time !